### ISSUE TITLE
Azure NC6 Tesla K80, Train hang after 5 minutes

### ISSUE THREAD

**Issue by tobads** [iperov/DeepFaceLab#188]
## Expected behavior
Train works.

## Actual behavior
![QQ截图20190313001424](https://user-images.githubusercontent.com/26805927/54217414-8ab98a80-4526-11e9-8432-9f670ad22bf4.jpg)
At first, train goes fine, but after 5 minutes, it just hang\stuck.
Program just hang there, always Epoch 127, wait 6 hours, nothing changed.
SAE, LIAE128, DF and so on, none of them works.

## Steps to reproduce
Put aligned face to dst and src, then run train.bat

## Other relevant information
Azure Cloud Platform
Instance:    Standard NC6 (1 Tesla K80 GPU)
Windows 10 1809
Tried Nvidia Driver 398.75 CUDA9.2 and 412.29 CUDA10. None of them works.
Tried  DeepFaceLabCUDA_build_28_02_2019,  DeepFaceLabCUDA_build_03_01_2019 RTX OK, DeepFaceLabCUDA9.0SSE_build_03_04_2019, DeepFaceLabCUDA9.2SSE_build_03_11_2019, DeepFaceLabCUDA10.1AVX_build_03_11_2019. None of them works.

Log Here:
<pre>Running trainer.

Loading model...

Model first run. Enter model options as default for each run.
Write preview history? (y/n ?:help skip:n/default) :
n
Target epoch (skip:unlimited/default) :
0
Batch_size (?:help skip:0/default) :
0
Feed faces to network sorted by yaw? (y/n ?:help skip:n) :
n
Flip faces randomly? (y/n ?:help skip:y) :
y
Src face scale modifier % ( -30...30, ?:help skip:0) :
0
Resolution ( 64-256 ?:help skip:128) :
128
Half or Full face? (h/f, ?:help skip:f) :
f
Learn mask? (y/n, ?:help skip:y) :
y
AE architecture (df, liae, vg ?:help skip:df) :
df
AutoEncoder dims (32-1024 ?:help skip:512) :
512
Encoder/Decoder dims per channel (21-85 ?:help skip:42) :
42
Use lightweight encoder? (y/n, ?:help skip:n) :
n
Use multiscale decoder? (y/n, ?:help skip:y) :
y
Use pixel loss? (y/n, ?:help skip: n/default ) :
n
Face style power ( 0.0 .. 100.0 ?:help skip:0.00) :
0.0
Background style power ( 0.0 .. 100.0 ?:help skip:0.00) :
0.0
D:\DeepFaceLabCUDA\_internal\bin\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from float to np.floating is deprecated. In future, it will be treated as np.float64 == np.dtype(float).type.
  from ._conv import register_converters as _register_converters
2019-03-12 16:07:05.803960: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2019-03-12 16:07:06.037394: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties:
name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235
pciBusID: 0001:00:00.0
totalMemory: 11.18GiB freeMemory: 10.94GiB
2019-03-12 16:07:06.043903: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-03-12 16:07:07.680015: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-12 16:07:07.683777: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0
2019-03-12 16:07:07.686102: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N
2019-03-12 16:07:07.691307: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10595 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0001:00:00.0, compute capability: 3.7)
Using TensorFlow backend.
WARNING:tensorflow:From D:\DeepFaceLabCUDA\_internal\bin\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From D:\DeepFaceLabCUDA\_internal\bin\lib\site-packages\tensorflow\python\ops\math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Loading: 100%|####################################################################| 1283/1283 [00:03<00:00, 421.90it/s]
Loading: 100%|####################################################################| 1540/1540 [00:04<00:00, 369.12it/s]
===== Model summary =====
== Model name: SAE
==
== Current epoch: 0
==
== Model options:
== |== batch_size : 4
== |== sort_by_yaw : False
== |== random_flip : True
== |== resolution : 128
== |== face_type : f
== |== learn_mask : True
== |== archi : df
== |== ae_dims : 512
== |== ed_ch_dims : 42
== |== lighter_encoder : False
== |== multiscale_decoder : True
== |== pixel_loss : False
== |== face_style_power : 0.0
== |== bg_style_power : 0.0
== Running on:
== |== [0 : Tesla K80]
=========================
Saving...
Starting. Press "Enter" to stop training and save model.
2019-03-12 16:07:35.149056: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library cublas64_100.dll locally
Saving...[#000165][1071ms] src_loss:4.926 dst_loss:9.92880</pre>

**Comment by tobads** [comment]
After inserting some log.
I found that program stuck in here:
```
_internal\python-3.6.8\Lib\site-packages\keras\backend\tensorflow_backend.py
def _call(self, inputs):
    fetched = self._callable_fn(*array_vals)    //stuck in this line
```
![3](https://user-images.githubusercontent.com/26805927/54469592-e0b35a00-47d4-11e9-964f-d5af76ff6463.png)
![1](https://user-images.githubusercontent.com/26805927/54469527-173ca500-47d4-11e9-8520-d6dc71bd6472.png)
![2](https://user-images.githubusercontent.com/26805927/54469529-17d53b80-47d4-11e9-9e67-359656afc9df.png)

@iperov 
Sir,you got any idea about this?
You can connect to my VPS if you want to, I will send you account and password.

**Comment by iperov** [comment]
STUCK ?
this is training function, of course it is stuck. lol

**Comment by iperov** [comment]
and I dont know how to fix. Tensorflow is not my code.
